dataflow/src/main/java/org/checkerframework/dataflow/analysis/FlowExpressions.java

```diff
@@ -315,6 +316,9 @@ public static Receiver internalReprOf(
                         receiver = null;
                 }
                 break;
+            case UNARY_PLUS:
+                return internalReprOf(
+                        provider, ((JCUnary) receiverTree).getExpression(), allowNonDeterministic);
```

Comment:
Instead of casting to the implementation class, it should be enough to use interface `com.sun.source.tree.UnaryTree`. Note how all other `tree` imports are of the interfaces, not the implementation classes.